### PR TITLE
Feature/abzu 180207 use cryptostream

### PR DESCRIPTION
**Pull request by JiviteshT**

### CODE REVIEW

**Review by HugoBurgess — APPROVED** [review]
Looks good

File: NVDSuppressions.xml [hunk 1]
```diff
@@ -85,4 +85,12 @@
 		</notes>
 		<cve>CVE-2022-39349</cve>
 	</suppress>
+	<suppress>
+		<notes>
+			<![CDATA[
+   file name: Microsoft.JScript.dll
```
Comment (rockydevnet) [inline NVDSuppressions.xml:91]: There should be some notes detailing why this suppression exits (likely a false positive based on the CVE being for a JavaScript thing???)

File: FileShareAdminClient/FileShareApiAdminClient.cs [hunk 2]
```diff
@@ -369,61 +372,63 @@ private async Task<IResult<AddFileToBatchResponse>> AddFiles(IBatchHandle batchH
...
-                        var putFileUri = $"batch/{batchHandle.BatchId}/files/{fileName}/{fileBlockIdAsString}";
-                        fileBlocks.Add(fileBlockIdAsString);
-                        ms.Seek(0, SeekOrigin.Begin);
+                            var fileBlockIdAsString = fileBlockId.ToString("D5");
+                            var putFileUri = $"batch/{batchHandle.BatchId}/files/{fileName}/{fileBlockIdAsString}";
+                            fileBlocks.Add(fileBlockIdAsString);
+                            ms.Seek(0, SeekOrigin.Begin);
 
-                        var blockMD5 = ms.CalculateMD5();
+                            var blockMD5 = ms.CalculateMD5();
```
Comment (rockydevnet) [inline FileShareAdminClient/FileShareApiAdminClient.cs:399]: Not sure this is correct, looks like you're still using the memory stream to calculate the MD5 hash?

**Review by rockydevnet — COMMENTED** [review]
Not sure that has achieved the goal of only streaming the source stream once?

File: FileShareAdminClient/FileShareApiAdminClient.cs (hunk 2 above)
Comment (rockydevnet) [reply FileShareAdminClient/FileShareApiAdminClient.cs:399]: Sorry, I take this back, it's calculating the block's MD5.

File: NVDSuppressions.xml (hunk 1 above)
Comment (JiviteshT) [reply NVDSuppressions.xml:91]: We regularly upgrade packages and go through suppressions to remove unnecessary entries. 
I will add notes and create user story to go through once upgraded to .net 8.

**Review by rockydevnet — APPROVED** [review]
LGTM

**Pull request merged**